sqlx-core/src/postgres/connection/mod.rs

```diff
@@ -100,6 +101,21 @@ impl PgConnection {
 
         Ok(())
     }
+
+    pub async fn server_version(&mut self) -> Result<String, Error> {
+        let result = self.fetch_one("SHOW server_version;").await?;
+        let server_version: String = result.get("server_version");
+
+        Ok(server_version)
+    }
+
+    pub async fn server_major_version(&mut self) -> Result<i32, Error> {
+        let server_version = self.server_version().await?;
+        let first = server_version.split(".").next().unwrap();
+        let major_version = first.parse::<i32>().unwrap();
+
+        Ok(major_version)
+    }
```

Comment:
We should actually be receiving this information already when establishing the connection, so we could just save this info in the connection when we receive it and then this can be a simple getter.

It'll be in `ParameterStatus` messages which the server sends unprompted on startup, and which we currently ignore: https://github.com/launchbadge/sqlx/blob/master/sqlx-core/src/postgres/connection/stream.rs#L107

https://www.postgresql.org/docs/13/protocol-flow.html#PROTOCOL-ASYNC

Reply:
Nice. It looks like this is the approach taken by @atkinschang in #1346. Do you want me to backport that into this PR, or wait for @atkinschang to clean this approach up in his rebased branch?

Reply:
I had him rebase on yours already so merging that should merge both.

Reply:
Great. I'll consider this resolved by #1346. Also happy to open another PR after #1346 if we need more cleanup to make sure there is only one (more efficient) way to do things.